refreshable/src/main/scala/prometheus4cats/refreshable/InstrumentedRefreshable.scala

```diff
@@ -386,39 +280,54 @@ object InstrumentedRefreshable {
           }
     }
 
-  def fromExisting[F[_]: MonadCancelThrow, A](
+  def fromExisting[F[_]: Async, A](
       refreshable: Refreshable[F, A],
       name: String,
       metricFactory: MetricFactory.WithCallbacks[F]
   ): Resource[F, InstrumentedRefreshable[F, A]] =
-    instanceMetrics[F, A](metricFactory).toResource
-//      .flatTap(_ => callback(name, refreshable, metricFactory))
-      .evalMap { case (readCounter, runningGauge, exhaustedRetriesGauge) =>
-        runningGauge
-          .set(true, name)
-          .as {
-            // Set the running gauge on read as we can't do this via callback or from the stream
-            val setRunningGauge = MonadCancelThrow[F].uncancelable { poll =>
-              poll(refreshable.get).flatTap(value =>
-                readCounter.inc(name -> value) >> (value match {
-                  case CachedValue.Cancelled(_) =>
-                    runningGauge.set(false, name)
-                  case _ => runningGauge.set(true, name)
-                })
+    (
+      instanceMetrics[F, A](metricFactory).toResource,
+      callbackMetrics(metricFactory).toResource
+    )
+      .flatMapN {
+        case (
+              (readCounter, runningGauge, exhaustedRetriesGauge),
+              (
+                refreshSuccessCounter,
+                currentlyFailingGauge,
+                refreshFailureCounter
               )
-
-            }
-
-            new InstrumentedRefreshable[F, A](
-              refreshable,
-              name,
-              readCounter,
-              runningGauge,
-              exhaustedRetriesGauge
-            ) {
-              override def get: F[CachedValue[A]] = setRunningGauge
+            ) =>
+//          callback(name, refreshable, metricFactory)
+//            .evalTap(_ => runningGauge.set(true, name))
+          runningGauge.set(true, name).toResource >> refreshable.updates
```

Comment:
I _think_ we want to wrap all of this in `Resource.uncancelable` to ensure that we can't be canceled during acquisition

Reply:
I suppose if this operation gets cancelled during acquisition that would reflect reality as the refreshable won't actually be running

Reply:
but I think it can happen after we set the running gauge to `true` but before we start the stream